src/html.rs

```diff
@@ -79,27 +87,40 @@ pub fn rewrite_html_base_url(
                 ),
                 // Rewrite background image url.
                 element!("body>div.bg-primary.text-main", |el| {
-                    if let Some(mut attr) = el.get_attribute("style") {
-                        if attr.starts_with("background-image: url('/") {
-                            let mut base_url = "";
-                            if let Some(url) = site_url {
-                                base_url = url;
+                    if let Some(style) = el.get_attribute("style") {
+                        let mut pairs = helpers::split_styles(&style);
+                        let backgrond_image_url = match pairs.get("background-image") {
+                            Some(value) if CSS_STATIC_URL_REGEX.is_match(value) => {
+                                if let Some(cdn_url) = cdn_url {
+                                    value.replacen("/static", cdn_url, 1)
+                                } else {
+                                    return Ok(());
+                                }
                             }
-                            if attr.starts_with("background-image: url('/static") {
-                                attr = attr.replacen("/static", "", 1);
-                                if let Some(url) = cdn_url {
-                                    base_url = url;
+                            Some(value) if CSS_ROOT_URL_REGEX.is_match(value) => {
+                                if let Some(site_url) = site_url {
+                                    value.replacen('/', &format!("{site_url}/"), 1)
```

Comment:
the replacen isn't an in-place modification, therefore it should be assigned the value

Reply:
Same as above